### PR TITLE
Setup CI jobs on Azure pipelines

### PR DESCRIPTION
**Pull request by nightlark**
### Summary

Setup commit and some daily jobs on Azure Pipelines

If merged this pull request will add Linux and macOS jobs to runs on Azure Pipelines, and some scheduled daily jobs for checking (possibly less common) variants.

### Proposed changes

- Add Linux and macOS jobs to Azure Pipelines
- Make some aspects of the CI build/test scripts use-able on services other than Travis CI

### CODE REVIEW

File: .ci/azure-pipelines-daily.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    branches:
+      include:
+        - helics3
+
+jobs:
+  - job: Daily Linux
+    strategy:
+      matrix:
+        all-tests:
+          containerImage: 'helics/buildenv:ubuntu18.04-default-builder'
```
Comment (phlptp) [inline .ci/azure-pipelines-daily.yml:13]: what compiler does the default builder use?

File: .ci/azure-pipelines-daily.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    timeoutInMinutes: 90
+
+    steps:
+      - bash: |
+          source scripts/setup-helics-ci-options.sh
+          mkdir -p build && cd build
+          ../scripts/ci-build.sh
+        env:
+          MAKEFLAGS: '-j 4'
+          CMAKE_GENERATOR: 'Unix Makefiles'
```
Comment (phlptp) [inline .ci/azure-pipelines-daily.yml:31]: should we have a test somewhere with the ninja generator?  Not that this needs to be.

Comment (nightlark) [reply .ci/azure-pipelines-daily.yml:31]: The builds on Drone CI use Ninja, but being able to have some of the tests use different generators in the future was one of the reasons for selecting the generator using an environment variable.

File: .ci/azure-pipelines-daily.yml (hunk 1 above)
Comment (nightlark) [reply .ci/azure-pipelines-daily.yml:13]: gcc7 - this is the build using the default version of all Ubuntu 18.04 packages.

**Pull request merged**